### PR TITLE
Air gap ruby gems

### PR DESCRIPTION
**Pull request by lilirui**
Add --local option to enforce install local ruby gems. 
Still WIP.

@schancel @tcahill @shr3kst3r @zquestz

### CODE REVIEW

**Comment by zquestz** [comment]
LGTM after squashing.

**Comment by schancel** [comment]
This looks correct -- can we do a test run on this prior to merge?

**Comment by lilirui** [comment]
I'm testing now.

File: recipes/default.rb [hunk 1]
```diff
@@ -26,18 +26,25 @@
   action :nothing
 end.run_action(:create)
 
-chef_gem "open4" do
+gem_package 'open4' do
   source File.join(Chef::Config['file_cache_path'], "open4.gem")
+  gem_binary Chef::Util::PathHelper.join(Chef::Config.embedded_dir,'bin','gem')
+  action :install
```
Comment (shr3kst3r) [inline recipes/default.rb]: Should this be `:update`?  If we were to update the minor version of the gem, the newer version would not get deployed, if I am reading the code correctly.  The `:install` would work if we were specifying a version in the gem_package().  From the docs for `:install`, ` Install a package. If a version is specified, install the specified version of the package.`

Comment (lilirui) [inline recipes/default.rb]: Do you mean ':upgrade' ? There is no update option.
```
:upgrade
Install a package and/or ensure that a package is the latest version. 
```
for air-gapped mode, we want to install the specific version we shipped only, so we should not use upgrade.

Comment (shr3kst3r) [inline recipes/default.rb]: Sorry, I am meant `:upgrade`.  But if the version1 is installed by release1, then when release2 comes out with a new version2, then `:install` will not upgrade.  We will get drift.

Comment (lilirui) [inline recipes/default.rb]: Done!  That makes sense.

Comment (shr3kst3r) [inline recipes/default.rb]: :+1:

**Comment by lilirui** [comment]
Deploy works at  http://tc.apcera.net/viewLog.html?buildId=105057&buildTypeId=SystemTestsApceraSetup_DeployTestBranchesAwsThreadedApceraSetup&tab=buildLog#_focus=608,
merging.

**Pull request merged**